### ISSUE TITLE
[Request] Add lint for unawaited coroutines

### ISSUE THREAD

**Issue by nrbnlulu** [microsoft/pyright#9086]
Currently you can safely do something like this
```py
class UserRepo:
	async def exists(self, pk: int) -> bool:
		return await self.session.exists(id=pk)
		

class UserService:
	repo: UserRepo
	
	async def delete_user(self, user: User) -> bool:
		if not self.repo.exists(user.id):
			return False
		await self.repo.delete(user.id)
		return True
```
As you can see this would always fail even if a user does exist because I forgot to await `self.repo.exists(user.id)`

Code sample in [pyright playground](https://pyright-play.net/?code=IYZwngdgxgBAJgUwGYyQezQCgJQwLQB8MAlhAC4BcAUDLTAE4JkCu9EMATDXVVaJLEQoARsHo58RAHJoICanRIp0WbAsV0ADvVJlMHbFQCQRoA)
___ 
Would be lovely if pyright would lint about this.

**Comment by erictraut** [comment]
Thanks for the suggestion. This is a duplicate of #3563. That issue has received zero thumbs-up reactions since it was posted, so this isn't (at least, so far) a very popular feature request. Please add your thumb-up to the other issue, and I'll consider it if and when there's signal from enough pyright users that it's something they'd like to see.

As I explain in the other issue, this check would need to be off by default — even in strict mode — because it would lead to false positive errors in legitimate Python code. I generally don't like adding new checks that cannot be enabled by default in strict mode because they're not very discoverable and therefore tend to get little or no usage from pyright users.

If you can think of a way to make this check safer — such that it's very unlikely to produce false positive errors, then it would be easier to justify adding such a feature.
